File: python/58.length-of-last-word.py
#
# @lc app=leetcode id=58 lang=python3
#
# [58] Length of Last Word
#

# @lc code=start
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        count = 0
        length = len(s)

        while length > 0 and s[length - 1] == ' ':
            length -= 1

        while length > 0 and s[length - 1] != ' ':
            count += 1
            length -=1
        
        return count

# @lc code=end

File: go/58.length-of-last-word.go
func lengthOfLastWord(s string) int {
    count:=0
    for i:= len(s)-1; i>= 0; i-- {
        if s[i]!=' ' {
            count++
        }else if s[i]==' ' && count>0 {
            return count
        }
    }
    return count
}